frontend/src/pages/QueryBuilder/QueryBuilder.tsx

```diff
@@ -74,30 +92,40 @@ class QueryBuilder extends React.Component<QueryBuilderProps> {
         </Col>
 
         <Col md={12} lg={8}>
-          <Card className={classNames({ 'd-none': !activeStep })}>
-            <Card.Header>
-              <Card.Title>
-                Create Query Step
-                <StyledIcon className="material-icons float-right" onClick={this.resetAction}>
-                  close
-                </StyledIcon>
-              </Card.Title>
-            </Card.Header>
-            <Card.Body>
-              <div className="mb-2">
-                {this.renderOperationStepForm(
-                  activeSource,
-                  activeStep,
-                  page.get('editingStep') as boolean,
-                )}
```

Comment:
I see why you created the card, but it was unnecessary ... you could have added your condition here for rendering either a preview or a step.

Reply:
Only difference besides this is a the title, and that's nothing a simple conditional can't fix.